Source/SwiftLintBuiltInRules/Rules/Style/TrailingClosureRule.swift

```diff
@@ -92,10 +166,91 @@ private extension FunctionCallExprSyntax {
         }
         return nil
     }
+
+    func dropLastArgument() -> Self {
+        let arguments = LabeledExprListSyntax(arguments.dropLast())
+
+        return self
+            .with(\.arguments, arguments.dropLastTrailingComma())
+            .dropParensIfEmpty()
+    }
+
+    func dropParensIfEmpty() -> Self {
+        if arguments.isEmpty {
+            self
+                .with(\.rightParen, nil)
+                .with(\.leftParen, nil)
+        } else {
+            self
+        }
+    }
+
+    func convertToTrailingClosure() -> Self? {
+        guard trailingClosure == nil, let lastDistinctClosureParameter else { return nil }
+        let leadingTrivia = lastTriviasInArguments?
+            .removingLeadingNewlinesIfExists()
+            .appendingSpaceIfNoTrailingSpace() ?? []
+
+        return dropLastArgument()
+            .with(\.trailingClosure, lastDistinctClosureParameter.with(\.leadingTrivia, leadingTrivia))
+            .with(\.calledExpression.trailingTrivia, [])
+    }
+
+    var lastTriviasInArguments: Trivia? {
+        guard let lastArgument = arguments.last,
+              let previous = lastArgument.previousToken(viewMode: .sourceAccurate)?.trailingTrivia else { return nil }
+
+        return previous
+            .merging(lastArgument.leadingTrivia)
+            .merging(triviaOf: lastArgument.label)
+            .merging(triviaOf: lastArgument.colon)
```

Comment:
There should also be tests for the colon and label trivia being merged.

Reply:
Fixed at eebb876.

Using `merging` here felt little awkward to me because it automatically removes common prefix and it might not be always what we want as users but at the same time, adding special case to handle whitespaces also seems too much? I would love your opinions on that. (Test has the both cases with and without the common trivia)

Reply:
Whitespace and comment handling feels always a little awkward. However, what I think is most important is that we don't delete comments, but keep them in the most reasonable way. Users can then go and fix or update them if necessary.

That said, merging the comments (and thereby removing common parts) feels like best we can do.